examples/call-runtime/lib.rs

```diff
@@ -0,0 +1,323 @@
+                }
+                _ => panic!(),
+            }
+        }
+    }
+
+    impl RuntimeCaller {
+        /// The constructor is `payable`, so that during instantiation it can be given some tokens
+        /// that will be further transferred with `transfer_through_runtime` message.
+        #[ink(constructor, payable)]
+        pub fn new() -> Self {
+            Default::default()
+        }
+
+        /// Tries to transfer `value` from the contract's balance to `receiver`.
+        ///
+        /// Fails if:
+        ///  - called in the off-chain environment
+        ///  - the chain doesn't allow `call-runtime` API (`UnsafeUnstableInterface` is turned off)
+        ///  - the chain forbids contracts to call `Balances::transfer` (`CallFilter` is too
+        ///    restrictive)
+        ///  - after the transfer, `receiver` doesn't have at least existential deposit  
+        ///  - the contract doesn't have enough balance
+        #[ink(message)]
+        pub fn transfer_through_runtime(
+            &mut self,
+            receiver: AccountId,
+            value: Balance,
+        ) -> Result<(), RuntimeError> {
+            self.env()
+                .call_runtime(&RuntimeCall::Balances(BalancesCall::Transfer {
+                    dest: receiver.into(),
+                    value,
+                }))
+                .map_err(Into::into)
+        }
+
+        /// Tries to trigger `call_runtime` API with rubbish data.
+        ///
+        /// # Note
+        ///
+        /// This message is for testing purposes only.
+        #[ink(message)]
+        pub fn call_nonexistent_extrinsic(&mut self) -> Result<(), RuntimeError> {
+            self.env().call_runtime(&()).map_err(Into::into)
+        }
+    }
+
+    #[cfg(all(test, feature = "e2e-tests"))]
+    mod e2e_tests {
+        use super::*;
+
+        use ink::{
+            env::{
+                test::default_accounts,
+                DefaultEnvironment,
+            },
+            primitives::AccountId,
+        };
+        use ink_e2e::build_message;
+
+        type E2EResult<T> = Result<T, Box<dyn std::error::Error>>;
+
+        /// The base number of indivisible units for balances on the `substrate-contracts-node`.
+        const UNIT: Balance = 1_000_000_000_000;
+
+        /// The contract will be given 1000 tokens during instantiation.
+        #[cfg(feature = "permissive-node")]
+        const CONTRACT_BALANCE: Balance = 1_000 * UNIT;
+
+        /// The receiver will get enough funds to have the required existential deposit.
+        ///
+        /// If your chain has this threshold higher, increase the transfer value.
+        const TRANSFER_VALUE: Balance = 1 / 10 * UNIT;
+
+        /// An amount that is below the existential deposit, so that a transfer to an empty account
+        /// fails.
+        ///
+        /// Must not be zero, because such an operation would be a successful no-op.
+        const INSUFFICIENT_TRANSFER_VALUE: Balance = 1;
```

Comment:
```suggestion
        /// Must not be zero, because such an operation would be a successful no-op.
        #[cfg(feature = "permissive-node")]
        const INSUFFICIENT_TRANSFER_VALUE: Balance = 1;
```